pyspades/server.py

```diff
@@ -332,17 +349,35 @@ def get_random_location(self, force_land=True, zone=(0, 0, 512, 512)):
 
     def set_master(self):
         if self.master:
-            get_master_connection(self).addCallbacks(
-                self.got_master_connection,
-                self.master_disconnected)
+            for game_version in sorted(self.game_versions):
+                if game_version not in self.master_connections:
+                    log.info('Adding master connection for version {!r}'.format(
+                        GAME_VERSION_TO_STRING[game_version],
+                    ))
+                    connection = get_master_connection(self, game_version)
+                    connection.addCallbacks(
+                        self.got_master_connection,
+                        self.master_disconnected)
 
     def got_master_connection(self, connection):
-        self.master_connection = connection
+        log.info('Got master connection for version {!r}'.format(
+            GAME_VERSION_TO_STRING[connection.game_version],
+        ))
+        self.master_connections[connection.game_version] = connection
```

Comment:
This could probably get a lot cleaner using `async/await`.

The restriction on only differentiating master connections only by version has obviously not aged that well.